### PR TITLE
STM32-H745 files for pysimCoder

### PR DESCRIPTION
**Pull request by robertobucher**
## Summary
Added files to use pysimCoder with the NUCLEO-H745ZI board

## Impact
 boards/arm/stm32h7/nucleo-h745zi/include/board.h
boards/arm/stm32h7/nucleo-h745zi/src/Makefile
boards/arm/stm32h7/nucleo-h745zi/src/nucleo-h745zi.h
boards/arm/stm32h7/nucleo-h745zi/src/stm32_bringup.c
boards/arm/stm32h7/nucleo-h745zi/configs/pysim/
boards/arm/stm32h7/nucleo-h745zi/src/stm32_adc.c
boards/arm/stm32h7/nucleo-h745zi/src/stm32_gpio.c
boards/arm/stm32h7/nucleo-h745zi/src/stm32_pwm.c
boards/arm/stm32h7/nucleo-h745zi/src/stm32_qencoder.c
boards/arm/stm32h7/nucleo-h745zi/src/stm32_usb.c

## Testing
Tested on NUCLEO-H745ZI-Q

### CODE REVIEW

File: boards/arm/stm32h7/nucleo-h745zi/src/stm32_adc.c [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+ ****************************************************************************/
+
+int stm32_adc_setup(void)
+{
+#if defined(CONFIG_STM32H7_ADC1) || defined(CONFIG_STM32H7_ADC3)
+  static bool initialized = false;
+  struct adc_dev_s *adc;
+  int ret;
+  int i;
+  char devname[] = "/dev/adc0";
```
Comment (acassis) [inline boards/arm/stm32h7/nucleo-h745zi/src/stm32_adc.c:138]: @robertobucher I suggest you to use devname[10] and:
```
snprintf(devname, 11, "/dev/adc%d", adcno);
```

Please look boards/arm/stm32/common/src/stm32_bmp180.c for reference

Comment (robertobucher) [reply boards/arm/stm32h7/nucleo-h745zi/src/stm32_adc.c]: Modified!

But what is the problem with the macOS (sim-01) build?

**Comment by acassis** [comment]
> How can I solve the problem with the Build / macOS (sim-02) (pull_request) ? Details are not usable to find out the problem...

@xiaoxiang781216  ?

**Comment by xiaoxiang781216** [comment]
@robertobucher could you squash your change into one patch?

**Pull request closed without merging**